### ISSUE TITLE
`S3Client.head_bucket` is annotated as returning nothing, but returns something

### ISSUE THREAD

**Issue by adamnovak** [youtype/mypy_boto3_builder#147]
**Describe the bug**

The stubs don't think `S3Client`'s `head_bucket` method returns anything. In fact, it returns a bunch of information about the bucket, but I get a typing error whenever I try and store it somewhere.

**To Reproduce**

1. Install `boto3-stubs[s3]`
2. Run `mypy`/`pyright` on the following code sample

```python
from mypy_boto3_s3 import S3Client
from typing import Any

import boto3

def do_head(c: S3Client) -> Any:
    info = c.head_bucket(Bucket='spacenet-dataset')
    return info

if __name__ == "__main__":
    print(do_head(boto3.client('s3')))

```

**Actual output**

```
repro.py: note: In function "do_head":
repro.py:7:12: error: "head_bucket" of "S3Client" does not return a value  [func-returns-value]
        info = c.head_bucket(Bucket='spacenet-dataset')
               ^
Found 1 error in 1 file (checked 1 source file)
```

**Expected output**

```
Success: no issues found in 1 source file
```

**Additional context**

This is on Linux, on Python 3.9, with these modules from `pip`:

```
pip list | grep boto
boto                          2.49.0
boto3                         1.21.16
boto3-stubs                   1.21.16
botocore                      1.24.16
botocore-stubs                1.23.38
mypy-boto3-iam                1.21.2
mypy-boto3-s3                 1.21.7
mypy-boto3-sdb                1.21.0
types-boto                    2.49.6
```

When I run that program, it prints:
```
{'ResponseMetadata': {'RequestId': '5YA3GFPAH2AX245Y', 'HostId': 'w3msLmINIKyK+v00Bb7omz84mP7uW+isshKkS0YaMoscpSY6ISZdyBnKzQhvCBi6JbyLdKiV9ag=', 'HTTPStatusCode': 200, 'HTTPHeaders': {'x-amz-id-2': 'w3msLmINIKyK+v00Bb7omz84mP7uW+isshKkS0YaMoscpSY6ISZdyBnKzQhvCBi6JbyLdKiV9ag=', 'x-amz-request-id': '5YA3GFPAH2AX245Y', 'date': 'Tue, 17 May 2022 22:40:06 GMT', 'x-amz-bucket-region': 'us-east-1', 'x-amz-access-point-alias': 'false', 'content-type': 'application/xml', 'server': 'AmazonS3'}, 'RetryAttempts': 1}}
```

**Comment by vemel** [comment]
Thank you for the report.

So, `boto3` documentation also says that the method returns `None`: https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/s3.html#S3.Client.head_bucket

However, even though the method response is empty, `boto3` always adds `ResponseMetadata` key to the output structure. 

A proper fix would be to change `None` return type to `{"ResponseMetadata": ...}` for all client methods that do not have output structure.

**Comment by vemel** [comment]
I decided to release the fix today. Tomorrow there should be a new minor release of `boto3`, so all services should be updated by tomorrow evening.

Please let me know if it works as it should.

**Comment by adamnovak** [comment]
This seems to have worked; our CI started failing my workaround code because it was an unnecessary ignore annotation, and my code now type-checks without it.
